common/environment/environment.cpp

```diff
@@ -1668,22 +1666,23 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
                     LOG(MCdebugInfo, unknownJob, "Drop zone coputer: '%s', IP: '%s'", dropZoneMachineName.str(), dropZoneMachineNetAddressIpString.str());
 #endif
                     if (targetIp.ipequals(dropZoneMachineIp))
-                       break;
-                    else
-                       // Path matched, but machine didn't
-                       dropZone = nullptr;
-                }
+                    {
+                        // Keep this drop zone
+                       if (dropzonePathLen > fullDropZoneDir.length())
```

Comment:
I was thinking about the 1st set, but ignoring fact that the initial value of dropzonePathLen is set at high max.

So yes, it's fine/correct as it is.